app/controllers/RepresentativeDanImportVATController.scala

```diff
@@ -59,38 +60,50 @@ class RepresentativeDanImportVATController @Inject()(identify: IdentifierAction,
         backLink
       ))),
       dan => {
-        val previousVATAccountNumber = request.userAnswers.get(AdditionalDefermentNumberPage).getOrElse(dan.accountNumber)
-        val previousVATAccountType = request.userAnswers.get(AdditionalDefermentTypePage).getOrElse(dan.danType)
-        if (dan.accountNumber != previousVATAccountNumber || dan.danType != previousVATAccountType) {
-
-          val authorityFiles = request.userAnswers.get(UploadAuthorityPage).getOrElse(Seq.empty).filterNot(_.dutyType == Vat)
-
-          for {
-            updatedAnswers <- Future.fromTry(request.userAnswers.set(UploadAuthorityPage, authorityFiles))
-            updatedAnswers <- Future.fromTry(updatedAnswers.set(CheckModePage, false))
-            updatedAnswers <- Future.fromTry(updatedAnswers.set(AdditionalDefermentTypePage, dan.danType))
-            updatedAnswers <- Future.fromTry(updatedAnswers.set(AdditionalDefermentNumberPage, dan.accountNumber))
-            _ <- sessionRepository.set(updatedAnswers)
+        val dutyAccountNumber = request.userAnswers.get(DefermentAccountPage).getOrElse("No Duty Account Number")
+        if (sameAccountNumber(dutyAccountNumber, dan.accountNumber, request.userAnswers)) {
```

Comment:
I'd probably avoid using `getOrElse` for this,
if we just want to compare the 2 numbers we can do it like
```suggestion
        val dutyAccountNumber = request.userAnswers.get(DefermentAccountPage)
        if (dutyAccountNumber.exists(sameAccountNumber(_, dan.accountNumber, request.userAnswers))) {
```